### ISSUE TITLE
Loss functions

### ISSUE THREAD

**Issue by soran-ghaderi** [soran-ghaderi/torchebm#16]
## GitHub Issue: Implement Loss Functions

This is my initial plan to implement as a set of loss functions. If you have any suggestions, feel free to add them here.

### Tasks

1. [ ] Contrastive Divergence (CD) Methods
   - [ ] Contrastive Divergence (CD-k)
   - [ ] Contrastive Divergence (PCD)
   - [ ] Persistent Contrastive Divergence (FPCD)
   - [ ] Parallel Tempering Contrastive Divergence (PTCD)

2. [ ] Score Matching Techniques
   - [ ] Standard Score Matching
   - [ ] Denoising Score Matching
   - [ ] Sliced Score Matching

3. [ ] Maximum Likelihood Estimation (MLE)

4. [ ] Margin Loss

5. [ ] Noise Contrastive Estimation (NCE)

6. [ ] Ratio Matching

7. [ ] Minimum Probability Flow

8. [ ] Adversarial Training Loss

9. [ ] Kullback-Leibler (KL) Divergence Loss

10. [ ] Fisher Divergence

11. [ ] Hinge Embedding Loss

12. [ ] Cross-Entropy Loss (for discrete outputs)

13. [ ] Mean Squared Error (MSE) Loss (for continuous outputs)

14. [ ] Improved Contrastive Divergence Loss

### Other

15. [ ] Unit tests
16. [ ] Documentation
17. [ ] Example usage
18. [ ] Benchmarking

### Acceptance Criteria
- All loss functions are implemented and working correctly
- Unit tests pass for all implementations
- Documentation is clear and comprehensive
- Example usage is provided for each loss function
- Code follows the library's style guide
